lib/haml/railtie.rb

```diff
@@ -26,7 +26,7 @@ class Railtie < ::Rails::Railtie
           require "haml/sass_rails_filter"
         end
 
-        if defined? Erubi
+        if defined?(::Erubi) && const_defined?('ActionView::Template::Handlers::ERB::Erubi')
```

Comment:
I prefer having only `const_defined?('ActionView::Template::Handlers::ERB::Erubi')`. safe_erubi_template.rb does not depend on the existence of `::Erubi`.